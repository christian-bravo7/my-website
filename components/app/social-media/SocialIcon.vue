<template>
  <a
    target="_blank"
    :href="link"
    class="w-10 h-10 bg-transparent dark:text-gray-50 dark-hover:bg-blue-500 dark-hover:border-blue-500 text-gray-900 hover:bg-pink-800 hover:text-gray-50 hover:border-pink-800 social-icon rounded-full transition-colors duration-200"
  >
    <slot />
  </a>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator';

@Component
export default class SocialIcon extends Vue {
  @Prop({ type: String, required: true })
  readonly link!: string;
}
</script>

<style lang="scss" scoped>
$light-color: #C5075D;
$dark-color: #5171FF;

.social-icon {
  padding: rem(10);

  &:hover {
    box-shadow: 0 0 0 5px rgba($light-color, 0.2);
  }

  svg {
    @apply fill-current;
  }
}

html.dark-mode {
  .social-icon {
    &:hover {
      box-shadow: 0 0 0 5px rgba($dark-color, 0.2);
    }
  }
}

</style>
