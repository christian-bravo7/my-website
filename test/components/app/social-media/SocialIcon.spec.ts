import { mount } from '@vue/test-utils';

import SocialIcon from '@/components/app/social-media/SocialIcon.vue';

describe('SocialIcon component', () => {
  it('should render correclty', () => {
    const wrapper = mount(SocialIcon, {
      slots: {
        default: '<svg></svg>',
      },
      propsData: {
        link: 'http://localhost/',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should display slot content', () => {
    const wrapper = mount(SocialIcon, {
      slots: {
        default: 'I am slot content',
      },
      propsData: {
        link: 'http://localhost/',
      },
    });

    const slotContent = wrapper.find('.social-icon');
    expect(slotContent.text()).toBe('I am slot content');
  });
});
