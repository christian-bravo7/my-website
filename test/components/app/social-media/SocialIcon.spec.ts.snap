// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialIcon component should render correclty 1`] = `<a target="_blank" href="http://localhost/" class="w-10 h-10 bg-transparent dark:text-gray-50 dark-hover:bg-blue-500 dark-hover:border-blue-500 text-gray-900 hover:bg-pink-800 hover:text-gray-50 hover:border-pink-800 social-icon rounded-full transition-colors duration-200"><svg></svg></a>`;
